### PR TITLE
 Implement watching component for IoT State Channels

### LINKED ISSUE #250: [IoT State Channels] Impl remote watching component
#### Motivation / Context

Provide remote watching interface

#### Description

Implement remote watching component with a protobuf over gRPC interface.

#### Testing

Test if the functionality works over gRPC interface.

#### Thread Safety

NA

### PR DESCRIPTION
**Pull request by manoranjith**
#### Description

 Implement watching component for IoT State Channels

Should be merged after #260 

##### Category

Implementation Task

##### Relevant issue

Closes #250 

#### Testing

Tests for this component has not yet been added. It will be done in a follow up task.
Until then, all linter checks and existing tests should pass.

##### Steps to run the tests

1. 
2. 
3.  

#### Checklist 

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

### CODE REVIEW

File: generateproto.go [hunk 1]
```diff
@@ -17,4 +17,5 @@
 package perun
 
 //go:generate protoc --proto_path=proto --go_out=api/grpc/pb --go-grpc_out=api/grpc/pb proto/nodetypes.proto proto/errors.proto proto/payment_service.proto
-//go:generate protoc --proto_path=proto --go_out=api/grpc/pb --go-grpc_out=api/grpc/pb proto/sdktypes.proto proto/funding_service.proto
+//go:generate protoc --proto_path=proto --go_out=api/grpc/pb --go-grpc_out=api/grpc/pb proto/sdktypes.proto proto/watching_service.proto
+//go:generate protoc --proto_path=proto --go_out=api/grpc/pb --go-grpc_out=api/grpc/pb proto/sdktypes.proto proto/watching_service.proto
```
Comment (danielksan81) [inline generateproto.go:21]: why is this entry doubled?

Comment (manoranjith) [reply generateproto.go:21]: I made a error while copying. Renamed both lines to watching_service, instead of just the copy

Comment (manoranjith) [reply generateproto.go:21]: Fixed in this force push: https://github.com/hyperledger-labs/perun-node/compare/b2a5067768ca2b1dc216ca42bbfbcba223f895a2..d5dcd874064ea43bb0f4192b23ff3f39f8665c6c

**Review by danielksan81 — APPROVED** [review]
lgtm

**Pull request merged**